### DISCUSSION TITLE
Transforming text nodes

### DISCUSSION THREAD

**Discussion by kamoshi** [markdoc/markdoc#365] [Q&A]
From reading the [documentation](https://markdoc.dev/docs/nodes#customizing-markdoc-nodes) it is not immediately obvious what should be returned from the transform method for text nodes.

```js
// inside markdoc config
nodes: {
    text: {
      transform(node, config) {
        return {} // ?
      }
    }
  }
```

In my use case, I would like to use [KaTeX](https://katex.org/) in particular to apply processing on arbitrary strings present inside a markdoc document. I would like to use it to render LaTeX as HTML/MathML embedded directly inside the output content.

Is something like this currently possible by using markdoc config?

**Comment by rpaul-stripe** [comment]
You can see the default behavior for the text node [here](https://github.com/markdoc/markdoc/blob/main/src/schema.ts#L218-L225). It returns `node.attributes.content` which is a string that contains the raw value of the text inside of the node. If you wanted to, you could do some processing on that string and return arbitrary render tree nodes instead.

If your KaTex syntax has any symbols that could be interpreted as Markdown, however, the parser could break it up into different nodes so you might not get the entire thing in a single text node. What I recommend doing instead is creating a custom Markdoc tag that performs KaTeX processing, so you could do something like:

~~~markdown
This is an example: {% katex "\\pm\\sqrt{a^2 + b^2}" %}
~~~

Note that you would unfortunately have to escape the slashes, since they are typically used for escape sequences in Markdoc strings.
